src/main/xerus/monstercat/downloader/Download.kt

```diff
@@ -129,9 +129,10 @@ class ReleaseDownload(private val release: Release, private var tracks: Collecti
 			totalProgress++
 		}
 		if(!isCancelled && downloadCover) {
-			val coverName = release.toString(COVERPATTERN()).replaceIllegalFileChars() + "." + release.coverUrl.substringAfterLast('.')
+			var coverName = release.toString(COVERPATTERN()).replaceIllegalFileChars()
 			updateMessage(coverName)
 			val entity = Covers.fetchCover(release.coverUrl, COVERARTSIZE())
+			coverName += "." + entity.contentType.value.substringAfter('/')
```

Comment:
you can just define the variable here and then move the updateMessage below it so it stays immutable, right?

Reply:
Right !

Reply:
5713a11a560484a92b7b392bf835054c6a89ac0b